### PR TITLE
Update changelog

### PR DESCRIPTION
**Pull request by gumb0**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,41 @@ Documentation of all notable changes to the **Fizzy** project.
 The format is based on [Keep a Changelog],
 and this project adheres to [Semantic Versioning].
 
+## [0.3.0] — Unreleased
+
```
Comment (axic) [inline CHANGELOG.md:9]: Need to add the usual summary of test results.

Comment (axic) [inline CHANGELOG.md:9]:
```suggestion
This main focus for this release is to implement every WebAssembly validation rule from the specification.

<include numbers>

Additionally measurable speed improvements were made for function calls:

<include benchmark>
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,52 @@ Documentation of all notable changes to the **Fizzy** project.
...
+  [#429](https://github.com/wasmx/fizzy/pull/429)
+- Type validation of branch instructions. [#408](https://github.com/wasmx/fizzy/pull/408)
+- Type validation of select instruction. [#413](https://github.com/wasmx/fizzy/pull/413)
+- Type validation in constant expressions. [#430](https://github.com/wasmx/fizzy/pull/430)
+- Validation of `else` branch in `if` instruction. [#417](https://github.com/wasmx/fizzy/pull/417)
+- Validation of frame types in `br_table` instruction. [#427](https://github.com/wasmx/fizzy/pull/427)
+- Validation of function indices in element section. [#432](https://github.com/wasmx/fizzy/pull/432)
+- Validation of functions' arity. [#433](https://github.com/wasmx/fizzy/pull/433)
+- Validation of alignment for floating point memory instructions. [#437](https://github.com/wasmx/fizzy/pull/437)
+- CI improvements. [#405](https://github.com/wasmx/fizzy/pull/405) [#434](https://github.com/wasmx/fizzy/pull/434)
```
Comment (chfast) [inline CHANGELOG.md:36]: Just drop it.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,52 @@ Documentation of all notable changes to the **Fizzy** project.
...
+- Better unit test coverage. [#381](https://github.com/wasmx/fizzy/pull/381) [#410](https://github.com/wasmx/fizzy/pull/410)
```
Comment (chfast) [inline CHANGELOG.md:37]: I would drop this as well. But if to stay, move it to `Changed`.

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,52 @@ Documentation of all notable changes to the **Fizzy** project.
...
+  [#431](https://github.com/wasmx/fizzy/pull/431) [#435](https://github.com/wasmx/fizzy/pull/435)
+
+### Changed
+- API cleanup (`execute` function overload taking `Module` is used only internally).
+  [#409](https://github.com/wasmx/fizzy/pull/409)
+- `execution_result` changed to returning single value instead of the entire stack and renamed to `ExecutionResult`. 
+  [#219](https://github.com/wasmx/fizzy/pull/219) [#428](https://github.com/wasmx/fizzy/pull/428)
+- Optimization of passing arguments to functions (eliminate extra copy). `execute` function now takes the arguments as 
+  `span<const uint64_t>`. [#359](https://github.com/wasmx/fizzy/pull/359) [#404](https://github.com/wasmx/fizzy/pull/404)  
+- Code style improvements. [#418](https://github.com/wasmx/fizzy/pull/418)
```
Comment (chfast) [inline CHANGELOG.md:47]: I would drop this one as well.

File: CHANGELOG.md (hunk 3 above)
Comment (axic) [reply CHANGELOG.md:37]: I think it is good keeping this under changed.

File: CHANGELOG.md (hunk 2 above)
Comment (axic) [reply CHANGELOG.md:36]: Actually, #405 added tests for C++20. That should be mentioned.

Comment (gumb0) [reply CHANGELOG.md]: Changed to mentioning C++20 tests only.

File: CHANGELOG.md (hunk 3 above)
Comment (gumb0) [reply CHANGELOG.md]: Moved to changed.

File: CHANGELOG.md (hunk 4 above)
Comment (gumb0) [reply CHANGELOG.md]: Dropped.

**Comment by gumb0** [comment]
Also added mentions of #354 and #444

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,55 @@ Documentation of all notable changes to the **Fizzy** project.
...
+## [0.3.0] — Unreleased
+
+This main focus for this release is to implement every WebAssembly validation rule from the specification.
+
+It passes a large part of the [official test suite (spectest 1.0)]:
+  - 4481 of 4490 binary parser and execution tests,
+  - 942 of 942 validation tests,
+  - 6381 skipped due to containing floating-point instructions or testing text format parser.
+
+Additionally measurable speed improvements were made for function calls:
```
Comment (axic) [inline CHANGELOG.md:17]:
```suggestion
Additionally, measurable speed improvements were made for branching and function calls:
```

Comment (gumb0) [reply CHANGELOG.md]: changed

**Comment by gumb0** [comment]
My results 0.2.0 va master
```
fizzy/execute/blake2b/512_bytes_rounds_1_median                   -0.0352         -0.0352           179           173           179           173
fizzy/execute/blake2b/512_bytes_rounds_16_median                  -0.0139         -0.0139          2642          2605          2642          2605
fizzy/execute/ecpairing/onepoint_median                           -0.0928         -0.0928       1000835        907948       1000774        907884
fizzy/execute/keccak256/512_bytes_rounds_1_median                 -0.0790         -0.0790           229           211           229           211
fizzy/execute/keccak256/512_bytes_rounds_16_median                -0.0644         -0.0643          3292          3080          3292          3080
fizzy/execute/memset/256_bytes_median                             -0.0225         -0.0224            14            14            14            14
fizzy/execute/memset/60000_bytes_median                           +0.0169         +0.0169          2944          2994          2944          2994
fizzy/execute/mul256_opt0/input0_median                           +0.0228         +0.0228            56            57            56            57
fizzy/execute/mul256_opt0/input1_median                           +0.0232         +0.0232            56            57            56            57
fizzy/execute/sha1/512_bytes_rounds_1_median                      -0.0312         -0.0312           184           178           184           178
fizzy/execute/sha1/512_bytes_rounds_16_median                     -0.0084         -0.0084          2504          2483          2504          2483
fizzy/execute/sha256/512_bytes_rounds_1_median                    -0.0282         -0.0282           176           171           176           171
fizzy/execute/sha256/512_bytes_rounds_16_median                   -0.0126         -0.0125          2364          2334          2364          2334
fizzy/execute/micro/eli_interpreter/halt_median                   -0.4434         -0.4434             0             0             0             0
fizzy/execute/micro/eli_interpreter/exec105_median                -0.0494         -0.0494             9             9             9             9
fizzy/execute/micro/factorial/10_median                           -0.3655         -0.3655             1             1             1             1
fizzy/execute/micro/factorial/20_median                           -0.3478         -0.3478             3             2             3             2
fizzy/execute/micro/fibonacci/24_median                           -0.3118         -0.3117         21949         15106         21947         15106
fizzy/execute/micro/host_adler32/1_median                         -0.5211         -0.5211             0             0             0             0
fizzy/execute/micro/host_adler32/100_median                       -0.5234         -0.5234            14             7            14             7
fizzy/execute/micro/host_adler32/1000_median                      -0.5231         -0.5231           135            64           135            64
fizzy/execute/micro/spinner/1_median                              -0.0407         -0.0406             0             0             0             0
fizzy/execute/micro/spinner/1000_median                           -0.0222         -0.0222            18            18            18            18

```

**Comment by chfast** [comment]
```
Comparing Fizzy 0.2.0 to 0.3.0 (Intel Haswell CPU, 4.0 GHz)
Benchmark                                  CPU Time [µs]       Old       New
----------------------------------------------------------------------------

fizzy/parse/blake2b                              +0.5960        14        23
fizzy/instantiate/blake2b                        +0.4412        19        28
fizzy/parse/ecpairing                            +0.8076       760      1373
fizzy/instantiate/ecpairing                      +0.7055       828      1412
fizzy/parse/keccak256                            +0.6071        26        42
fizzy/instantiate/keccak256                      +0.4818        32        47
fizzy/parse/memset                               +0.3710         4         6
fizzy/instantiate/memset                         +0.2013         8        10
fizzy/parse/mul256_opt0                          +0.6683         5         8
fizzy/instantiate/mul256_opt0                    +0.3370         9        12
fizzy/parse/sha1                                 +0.6466        23        38
fizzy/instantiate/sha1                           +0.4696        29        43
fizzy/parse/sha256                               +0.7765        36        63
fizzy/instantiate/sha256                         +0.6920        42        70
fizzy/parse/micro/eli_interpreter                +0.2003         3         4
fizzy/instantiate/micro/eli_interpreter          +0.0935         7         8
fizzy/parse/micro/factorial                      +0.2105         1         1
fizzy/instantiate/micro/factorial                +0.1369         1         1
fizzy/parse/micro/fibonacci                      +0.2773         1         1
fizzy/instantiate/micro/fibonacci                +0.1888         1         2
fizzy/parse/micro/host_adler32                   +0.1448         1         2
fizzy/instantiate/micro/host_adler32             +0.0674         4         4
fizzy/parse/micro/spinner                        +0.1232         1         1
fizzy/instantiate/micro/spinner                  +0.0774         1         1

fizzy/execute/blake2b/512_bytes_rounds_1         -0.0250        86        83
fizzy/execute/blake2b/512_bytes_rounds_16        -0.0197      1294      1268
fizzy/execute/ecpairing/onepoint                 -0.1027    479448    430202
fizzy/execute/keccak256/512_bytes_rounds_1       -0.0422       104       100
fizzy/execute/keccak256/512_bytes_rounds_16      -0.0352      1512      1458
fizzy/execute/memset/256_bytes                   -0.0318         7         7
fizzy/execute/memset/60000_bytes                 -0.0109      1609      1592
fizzy/execute/mul256_opt0/input0                 -0.0254        27        26
fizzy/execute/mul256_opt0/input1                 -0.0262        27        26
fizzy/execute/sha1/512_bytes_rounds_1            -0.0452        95        90
fizzy/execute/sha1/512_bytes_rounds_16           -0.0478      1318      1255
fizzy/execute/sha256/512_bytes_rounds_1          +0.0446        93        97
fizzy/execute/sha256/512_bytes_rounds_16         +0.0457      1280      1338
fizzy/execute/micro/eli_interpreter/halt         -0.3601         0         0
fizzy/execute/micro/eli_interpreter/exec105      -0.0299         5         5
fizzy/execute/micro/factorial/10                 -0.2383         1         0
fizzy/execute/micro/factorial/20                 -0.2363         1         1
fizzy/execute/micro/fibonacci/24                 -0.2760     10273      7438
fizzy/execute/micro/host_adler32/1               -0.5324         0         0
fizzy/execute/micro/host_adler32/100             -0.5275         6         3
fizzy/execute/micro/host_adler32/1000            -0.5303        63        30
fizzy/execute/micro/spinner/1                    -0.0677         0         0
fizzy/execute/micro/spinner/1000                 -0.0256        10        10
```

**Pull request merged**